src/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableQueue_1.cs

```diff
@@ -141,6 +141,23 @@ public T Peek()
             return _forwards.Peek();
         }
 
+#if FEATURE_ITEMREFAPI
+        /// <summary>
+        /// Gets a read-only reference to the element at the front of the queue.
+        /// </summary>
+        /// <exception cref="InvalidOperationException">Thrown when the queue is empty.</exception>
+        [Pure]
```

Comment:
What is PureAttribute for? it brought my attention that it is used only in some APIs.

Reply:
I followed the pattern that exists already in the code base, as only some indexer APIs are annotated with `[Pure]`. We can do a later analysis to annotate all applicable methods if we want.

Reply:
Ok, sounds good.